include/spm/rel/an2_08.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <common.h>
+#include <spm/evtmgr.h>
+#include <evt_cmd.h>
+
+CPP_WRAPPER(spm::an2_08)
+
+typedef struct RpgNPC {
+/* 0x00 */ s32 flags;
+/* 0x04 */ s32 unk4;
+/* 0x08 */ s32 maxHp;
+/* 0x0C */ s32 attackStrength;
+/* 0x10 */ s32 unk_10;
+/* 0x14 */ s32 killXp;
+} RpgUnk;
+SIZE_ASSERT(RpgNPC, 0x18);
+
+typedef struct RpgMenu {
+/* 0x00 */ s32 flags;
+/* 0x04 */ s32 unk_04;
+/* 0x08 */ void* unk_08;
+/* 0x0C */ RpgNPC rpgNpcInfo[3];
+/* 0x54 */ s32 unk_54;
+/* 0x58 */ s32 unk_58[15];
+/* 0x94 */ char unk_94[4];
+} RpgMenu; //sizeof 0x98
+SIZE_ASSERT(RpgMenu, 0x98);
+
+typedef struct PhysicalMenu {
```

Comment:
What does this name actually mean?

Reply:
that struct determines the actual physical menu of the turn based combat, by filling the struct with things like Red Underchomp (for list of things you can attack) as well as the general combat option list

Reply:
Maybe it'd be better if RpgMenu was the name for this struct rather than the other one then? Since the other seems to be more controlling the rpg logic than the menu itself.

Asserts in FUN_80c6cccc seem like the other struct is just the work struct for an2_08 (these are handled in spm-headers by wrapping them in DECOMP_STATIC and giving them a snake case prefix of their filename like https://github.com/SeekyCt/spm-headers/blob/3628fc4fb33ce8d9b8af56cc77c571f48c99c3ee/include/spm/dvdmgr.h#L36)

Reply:
The second paragraph doesn't make sense to me, memory is allocated for the RpgMenu struct inside a user func only called in the rpg battle (evt_rpg_npctribe_handle) and immediately fills it with the underchomp's stats, so unless I'm misunderstanding just what a work struct is, I don't think that would be it

The first paragraph does make sense to me, but we should settle what the name of the original RpgMenu should be before I rename what is currently

On a positive note though, FUN_80c6cccc makes it seem like unk_08 of RpgMenu is just a pointer to a PhysicalMenu struct, what are your thoughts on that?

Reply:
correct me if I'm wrong but the ivar is unk_04 from the RpgMenu struct, wouldn't that mean that unk_04 is a pointer to the wp rather than RpgMenu itself being the wp, since unk_04 is what's being asserted?

Reply:
Also RpgMenu is only 0x98 long, the assert says 0x131

Reply:
0x131 is the line number of the assert, not struct size
Ghidra is reversing the comparison, it's equivalent to `i < rpg_menu->unk_04` which the assert shows was originally `i < wp->num`

Reply:
im still a little confused but i'll just take your word for it lmao
what should I rename it to?

Reply:
The assert expands from this macro https://github.com/SeekyCt/spm-headers/blob/3628fc4fb33ce8d9b8af56cc77c571f48c99c3ee/include/spm/system.h#L23
So it would've originally been `assert(i < wp->num, "error")`. __assert2 is the error handling function called when an assert fails, with context like filename and line number.

The variable can become an2_08_wp and the struct something like An2_08Work